gui/js/onboarding.window.js

```diff
@@ -42,9 +43,11 @@ module.exports = class OnboardingWM extends WindowManager {
     // through ports so we can trigger 'registration-done' without relying
     // on timeouts
     this.send('registration-done')
-    this.win.once('dom-ready', () => {
-      setTimeout(() => this.send('registration-done')
-      , 20)
+    this.win.webContents.once('dom-ready', () => {
+      setTimeout(() => {
+        this.send('registration-done')
+        this.checkSyncPath(defaults.syncPath, this)
```

Comment:
Using `this` as a `eventSender` is kind of hacky.

Reply:
Agreed.

I was feeling wrong about passing `this.win.webContents` without first checking it still exists, which `this.send()` already does (well actually `this.win` could be `null`, but `this.win` is not supposed to exist without `this.win.webContents` since `this.win` is a `BrowserWindow`...).

Another solution could be to pass a `send` function...

What were you thinking of?

Reply:
I'am thinking this is hacky, but still a lot less than the whole dom-ready+timeout that surrounds it. Maybe just a `// XXX` to be considered if we ever get around o fix `elm<->ports<->renderer<->main` pipeline.